Can't used a fixed sized array as a return value

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** 
**OS:** linux - Using C backend

**What did you do?**
Try to use fixed sized array as return type

**What did you expect to see?**
Program compile and run
 
**What did you see instead?**
C compiler error
```
/tmp/v/36cube.18105555941475086726.tmp.c: In function ‘main__main’:
/tmp/v/36cube.18105555941475086726.tmp.c:10169:42: error: lvalue required as unary ‘&’ operand
  memcpy((Array_fixed_int_6*)nums, (byte*)&main__set_value(nums, 0, 1), sizeof(Array_fixed_int_6));
                                          ^
/tmp/v/36cube.18105555941475086726.tmp.c: At top level:
/tmp/v/36cube.18105555941475086726.tmp.c:10173:35: error: ‘main__set_value’ declared as function returning an array
 VV_LOCAL_SYMBOL Array_fixed_int_6 main__set_value(Array_fixed_int_6 nums, int i, int v) {
                                   ^~~~~~~~~~~~~~~
/tmp/v/36cube.18105555941475086726.tmp.c: In function ‘main__set_value’:
/tmp/v/36cube.18105555941475086726.tmp.c:10177:9: warning: returning ‘int *’ from a function with return type ‘int’ makes integer from pointer without a cast [-Wint-conversion]
  return n;
         ^
...
```

So interestingly on x86_64 this compiles and runs with the native backend, albeit it initializes the starting of the array with a bunch of wacky values.  However I am doing embedded development, and with the the C backend I get the above error on different architectures.

Example:
```
fn main() {
        mut nums := [6]int{}
        nums = set_value(nums, 0, 1)
        println(nums)
}

fn set_value(nums [6]int, i int, v int) [6]int {
        mut n := nums
        n[i] = v
        return n
}
```
And understandably fixed sized multidimensional arrays don't compile either.
```
fn main() {
        mut nums := [6][6]int{}
        nums = set_value(nums, 0, 1)
        println(nums)
}

fn set_value(nums [6][6]int, i int, v int) [6][6]int {
        mut n := nums
        n[i][i] = v
        return n
}
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -7790,17 +7790,21 @@ fn (mut c Checker) fn_decl(mut node ast.FnDecl) {
 		return_sym := c.table.get_type_symbol(node.return_type)
 		if return_sym.info is ast.MultiReturn {
 			for multi_type in return_sym.info.types {
+				multi_sym := c.table.get_type_symbol(multi_type)
 				if multi_type == ast.error_type {
 					c.error('type `IError` cannot be used in multi-return, return an option instead',
 						node.return_type_pos)
 				} else if multi_type.has_flag(.optional) {
 					c.error('option cannot be used in multi-return, return an option instead',
 						node.return_type_pos)
+				} else if multi_sym.kind == .array_fixed {
+					c.error('fixed array cannot be used in multi-return', node.return_type_pos)
 				}
 			}
+		} else if return_sym.kind == .array_fixed {
+			c.error('fixed array cannot be returned by function', node.return_type_pos)
 		}
-	}
-	if node.return_type == ast.void_type {
+	} else {
 		for mut a in node.attrs {
 			if a.kind == .comptime_define {
 				c.evaluate_once_comptime_if_attribute(mut a)

--- a/vlib/v/checker/tests/return_fixed_array.out
+++ b/vlib/v/checker/tests/return_fixed_array.out
@@ -1,0 +1,12 @@
+vlib/v/checker/tests/return_fixed_array.vv:1:25: error: fixed array cannot be returned by function
+    1 | fn return_fixed_array() [3]int {
+      |                         ~~~~~~
+    2 |     return [1, 2, 3]!
+    3 | }
+vlib/v/checker/tests/return_fixed_array.vv:5:41: error: fixed array cannot be used in multi-return
+    3 | }
+    4 | 
+    5 | fn return_fixed_array_in_multi_return() ([3]int, [3]int) {
+      |                                         ~~~~~~~~~~~~~~~~
+    6 |     return [1, 2, 3]!, [4, 5, 6]!
+    7 | }

--- a/vlib/v/checker/tests/return_fixed_array.vv
+++ b/vlib/v/checker/tests/return_fixed_array.vv
@@ -1,0 +1,7 @@
+fn return_fixed_array() [3]int {
+	return [1, 2, 3]!
+}
+
+fn return_fixed_array_in_multi_return() ([3]int, [3]int) {
+	return [1, 2, 3]!, [4, 5, 6]!
+}

